python/ray/data/tests/test_dataset_iterator.py

```diff
@@ -37,6 +37,16 @@ def test_basic_dataset(ray_start_regular_shared):
     # assert it.stats() == ds.stats()
 
 
+def test_basic_dataset_iter_rows(ray_start_regular_shared):
+    ds = ray.data.range(100)
+    it = ds.iterator()
+    for _ in range(3):
```

Comment:
Why're we repeating this 3 times?

Reply:
This is what we want to test- that we can iterate through the iterator multiple times.

Reply:
Gotcha.

Super pedantic, but if there's nothing special about 3, it might be better to use 2: https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#write-minimally-passing-tests